### ISSUE TITLE
[Enhancement]: Admin Reports - Allow custom orderby 

### ISSUE THREAD

**Issue by GlennTaylorDigital** [woothemes/woocommerce#46275]
### Describe the solution you'd like

Hi team.

I'm working on a project where I've added some count values to the customers report screen so we can see how many reports we've had for things such as orders that didn't arrive, arrived late, arrived damaged etc. These values are cached to the user meta table, then displayed on the admin report so the client can easily find repeat offenders etc.

To improve this I want allow them to sort by the column, so they can easily see who has the most order issues. To do that I've added 

`isSortable: true`

To the column, which allows me to click to sort the column, but I then receive a 400 error with the following response:

```
{
    "code": "rest_invalid_param",
    "message": "Invalid parameter(s): orderby",
    "data": {
        "status": 400,
        "params": {
            "orderby": "orderby is not one of username, name, country, city, state, postcode, date_registered, date_last_active, orders_count, total_spend and avg_order_value."
        },
        "details": {
            "orderby": {
                "code": "rest_not_in_enum",
                "message": "orderby is not one of username, name, country, city, state, postcode, date_registered, date_last_active, orders_count, total_spend and avg_order_value.",
                "data": null
            }
        }
    }
}
```

To resolve this, it would be fantastic if we could get a filter wrapped around the collection params for each of the Reports. I can get the custom sort to work by modifying src/Admin/API/Reports/Customers and adding a new array entry below line 333 (avg_order_value) under $params['orderby']['enum'] but obviously I don't want to change the core plugin!

If I've missed anything, please let me know.

Thanks in advance.

**Comment by webaxones** [comment]
Hi,
Is anyone working on this? Because allowing custom product meta or product attributes to be added to reports but not allowing them to be sorted means removing 50% of the potential. Examples that I am coding: product year, or waiting list counter. I should be able to do an ORDER BY for the 2. For variations there is an `experimental_woocommerce_analytics_variations_collection_params` but it is limited to the variation report and using an experimental in production is not possible.
Please try adding this/these filters: it would add a lot to the reports. :)

**Comment by ianlin** [comment]
This enhancement has been addressed in #49722, and it will be included in the next WooCommerce `9.4`.
